Source/JavaScriptCore/dfg/DFGByteCodeParser.cpp

```diff
@@ -5191,15 +5194,37 @@ void ByteCodeParser::handleGetById(
 
     // Special path for custom accessors since custom's offset does not have any meanings.
     // So, this is completely different from Simple one. But we have a chance to optimize it when we use DOMJIT.
-    if (Options::useDOMJIT() && getByStatus.isCustom()) {
-        ASSERT(getByStatus.numVariants() == 1);
-        ASSERT(!getByStatus.makesCalls());
+    if (getByStatus.numVariants() == 1) {
         GetByVariant variant = getByStatus[0];
-        ASSERT(variant.domAttribute());
-        if (handleDOMJITGetter(destination, variant, base, identifierNumber, prediction)) {
+        if (getByStatus.isCustom()) {
+            // DOMGetter does not perform type check for base. So if we found variant.domAttribute(), we must use CallDOMGetter.
+            if (Options::useDOMJIT() && variant.domAttribute()) {
+                ASSERT(!getByStatus.makesCalls());
+                if (handleDOMJITGetter(destination, variant, base, identifierNumber, prediction)) {
+                    if (UNLIKELY(m_graph.compilation()))
```

Comment:
Wow, `compilation` is a not a good function name for the Bytecode profile's compilation object...